lib/srv/db/cloud/iam.go

```diff
@@ -142,6 +142,7 @@ func (c *IAM) Start(ctx context.Context) error {
 // Setup sets up cloud IAM policies for the provided database.
 func (c *IAM) Setup(ctx context.Context, database types.Database) error {
 	if c.isSetupRequiredForDatabase(database) {
+		c.iamPolicyStatus.Store(database.GetName(), types.IAMPolicyStatus_IAM_POLICY_STATUS_PENDING)
```

Comment:
Do we even need to set it to "pending"? If it's not "success" or "failure" then it's basically "pending", no?

Reply:
perhaps i'm wrong... but I needed `pending` to help me differentiate "service running old teleport" vs "service running new teleport".

A service can be running an older version of teleport that does not set the marker that the client will be looking for, and the flag will always be "unspecified" (or some other zero value if we go the string route). 

Newer agent (this PR), will set it to "pending".

How it is used in the web UI is like this:

- if the value the client gets is "unspecified", we stop polling right away b/c i know that if it's "unspecified", it is an old svc, and this value will never change. This unfortunately means we can't tell if we can skip the IAM policy screen or not, but I didn't think it was too big of a deal (I can add a skip button and indicate that it may be optional)

- if the value we get is "pending", we will keep polling until that "pending" changes to another value (error, invalid, valid), and then depending on the value we get, we can skip the IAM policy.

Reply:
@kimlisa Gotcha, that's fine then. The only suggestion I have is to have some sort of (short) timeout when polling for pending status.